service/src/main.rs

```diff
@@ -125,13 +132,17 @@ fn main() {
 	let tokio_handle = Arc::new(GlobalTokioHandle {});
 	let sync_block_gossiper =
 		Arc::new(SyncBlockGossiper::new(tokio_handle.clone(), worker.clone()));
-	let peer_updater = Arc::new(GlobalPeerUpdater::new(worker));
+	let peer_updater = Arc::new(WorkerPeersUpdater::new(worker));
 	let sidechain_blockstorage = Arc::new(
 		SidechainStorageLock::<SignedSidechainBlock>::new(PathBuf::from(&SIDECHAIN_STORAGE_PATH))
 			.unwrap(),
 	);
-	let node_api_factory = Arc::new(GlobalUrlNodeApiFactory::new(config.node_url()));
+	let node_api_factory =
+		Arc::new(NodeApiFactory::new(config.node_url(), AccountKeyring::Alice.pair()));
```

Comment:
The signer is now passed into the node API factory and doesn't have to be set anymore as an additional step (is included in the factory now)